internal/keycard_context_v2.go

```diff
@@ -565,18 +569,27 @@ func (kc *KeycardContextV2) onAuthorizeInteractions(authorized bool) {
 	kc.publishStatus()
 }
 
-func (kc *KeycardContextV2) VerifyPIN(pin string) (err error) {
-	if err = kc.keycardReady(); err != nil {
-		return err
+func (kc *KeycardContextV2) VerifyPIN(pin string) (err error, authorized bool) {
+	if err := kc.keycardReady(); err != nil {
+		return err, false
 	}
 
 	defer func() {
-		authorized := err == nil
+		authorized := err == nil && authorized
```

Comment:
I think it'd be better to either rename the `authorized` inside the closure to something else or remove the `:` depending if you want or not to modify the actual return value. The way it is now it's a bit confusing to read imho

Reply:
actually wondering if this line is needed anymore at all. I can't see any case where `authorized` would be true but `err` is not `nil`